agent/agent.go

```diff
@@ -180,28 +168,27 @@ func (a *Agent) registerConsulService() error {
 // is reported back in the "discover_cluster" Service in consul under a TTL of 60 minutes
 func (a *Agent) startDiscoverTicker() {
 	tick := func() {
-		for _, d := range a.discoveries {
-			var status, result string
-			var err error
+		var note []string
+		status := consulApi.HealthPassing
 
-			result, err = d.Discover()
+		for _, d := range a.discoveries {
+			result, err := d.Discover()
 			if err != nil {
-				log.Printf("Error while running discovery '%s': %s", d.GetId(), err)
-				result = err.Error()
-				status = consulApi.HealthWarning
-			} else {
-				status = consulApi.HealthPassing
-			}
+				result = fmt.Sprintf("Error while running discovery '%s': %s", d.GetId(), err)
+				status = consulApi.HealthCritical
 
-			err = a.consul.Agent().UpdateTTL(d.GetId(), result, status)
-			if err != nil {
-				log.Println("An error occurred while trying to update TTL with Consul:", err)
+				log.Errorln(result)
 			}
+			note = append(note, result)
```

Comment:
nitpick: `note` is _very_ ambiguous as a variable name here, because the consul `HealthCheck` structure has a `Notes` field, but what we're actually populating later with the `UpdateTTL()` function is the `Output` field.

Reply:
I see! changed to `output`

I was using note because the signature of UpdateTTL() calls it so, but i see the confusion also inside the consul API since the check definition has a note field as well
![image](https://user-images.githubusercontent.com/828651/133295083-af6a97f4-b1f5-40b1-806a-934bf7a0a481.png)

Reply:
maybe `resultNote`? `discoveryNote`?

Reply:
Oh, `output` is fine